sqlmesh/core/context_diff.py

```diff
@@ -42,26 +42,26 @@ class ContextDiff(PydanticModel):
     """Whether the currently stored environment record is in unfinalized state."""
     create_from: str
     """The name of the environment the target environment will be created from if new."""
-    added: t.Set[str]
+    added: t.Set[SnapshotId]
     """New nodes."""
-    removed_snapshots: t.Dict[str, SnapshotTableInfo]
+    removed_snapshots: t.Dict[SnapshotId, SnapshotTableInfo]
     """Deleted nodes."""
-    modified_snapshots: t.Dict[str, t.Tuple[Snapshot, Snapshot]]
+    modified_snapshots: t.Dict[SnapshotId, t.Tuple[Snapshot, Snapshot]]
```

Comment:
Somehow this mapping doesn't make much sense to me. We have 2 snapshots that represent a change. Which snapshot ID is used as a key then? Why that snapshot ID and not the other one?

This is very counterintuitive. What's worth is that this confusion spreads further into methods below: `directly_modified`, etc. As a caller how do I suppose to know whether to pass the source Snapshot ID or the destination one?